department-of-veterans-affairs/caseflow
# APPEALS-8476

Resolves [APPEALS-8476](https://vajira.max.gov/browse/APPEALS-8476) 

### Description
- Added feature toggles to point to all_case_distribution.rb
- Added methods to get {limit} number of nonpriority appeals from all dockets, sort by age only, then get the oldest {limit} number of those appeals and distribute them for tied legacy and all non-tied appeals
- Added BFDNOD (Notice of Disagreement date; correlated to receipt_date in Caseflow) to the VACOLS SQL queries since that is the date we should be sorting by
- Added condition “is not null” for BFAC/BFDNOD in SELECT_READY_APPEALS to prevent the query from getting cases from VACOLS with incomplete data that shouldn’t be counted. Without this, the cases would be used in the above methods and selected as ready, but without those columns will not be selected by the distribution queries. This would cause the request_distribution to eventually distribute nothing, since the same VACOLS cases would take up the spots as the oldest appeals but would never be distributed properly
- Modified several seed files to ensure that VACOLS cases get created with values for BFAC and BFDNOD so that testable data is seeded; without this change, zero nonpriority genpop cases would be seeded in VACOLS. Additionally, modified seeds so that NOD/receipt dates are not all set to the time of seeding
- Added file spec/models/concerns/all_case_distribution_spec.rb for tests

### Acceptance Criteria
- [ ] Spec file passes
- [ ] Running a request distribution from a judge in Caseflow distributes nonpriority cases based only on age instead of proportions

### Testing Plan
1. If running locally, run 'make reset' to reseed the DB, then run the frontend and backend
2. Go to Caseflow
3. Select a judge user who can be assigned cases (BVAEBECKE, BVAAABSHIRE, **BVABDANIEL** are three examples)
4. Go to queue and switch views to the 'Assign team cases' view
5. If cases are available to assign, assign them to attorneys until there are less than three awaiting assignment
6. Click 'Request more cases'
7. Ensure the 'Distribution Complete' banner appears and distributed cases populate on the page
  - Assuming the judge has no tied priority cases **(BVABDANIEL in my testing)**, seed data on this branch will yield 4 legacy appeals and 5 direct review on the first request, then 7 direct review and 2 evidence review on the second request, then 5 direct review and 4 evidence review on the third request
